pkg/autodiscover/autodiscover_podset.go

```diff
@@ -71,54 +82,65 @@ func findDeploymentByLabel(
 		if len(dps.Items) == 0 {
 			log.Warn("Did not find any deployments in ns=%s", ns)
 		}
-
 		for i := 0; i < len(dps.Items); i++ {
-			for _, aLabelObject := range labels {
-				log.Debug("Searching pods in deployment %q found in ns %q using label %s=%s", dps.Items[i].Name, ns, aLabelObject.LabelKey, aLabelObject.LabelValue)
-				if dps.Items[i].Spec.Template.ObjectMeta.Labels[aLabelObject.LabelKey] == aLabelObject.LabelValue {
-					deployments = append(deployments, dps.Items[i])
-					log.Info("Deployment %s found in ns=%s", dps.Items[i].Name, ns)
-				}
+			if len(labels) > 0 {
+				allDeployments = findDeploymentsMatchingAtLeastOneLabel(labels, ns, &dps.Items[i], allDeployments)
```

Comment:
Looks like there is a bug in the original code... If a pod in the deployment matches multiple labels, the deployment will be added multiple times in the list. We should add the deployment once if at least one pod matches one label in the deployment/statefulset. see https://github.com/shimritproj/cnf-certification-test/pull/120

Reply:
Thank you for your attention :) 
I fixed it in Deployment and StatefulSet.